frontend/js/src/common/UserSearch.tsx

```diff
@@ -14,97 +11,131 @@ export type UserSearchProps = {
-    const { APIService } = this.context;
-    const { newUser } = this.state;
+  const searchUsers = async () => {
     try {
       const response = await APIService.searchUsers(newUser);
-      this.setState({
-        userSearchResults: response.users,
-      });
+      setUserSearchResults(response.users);
     } catch (error) {
-      this.handleError(error);
+      toast.error(<ToastMsg title="Error" message={error.message} />, {
+        toastId: "error",
+      });
     }
   };
 
-  handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState(
-      {
-        newUser: event.target.value,
-      },
-      this.throttledSearchUsers
-    );
-  };
+  const throttledSearchUsers = _throttle(async () => {
+    await searchUsers();
+  }, 300);
 
-  handleResultClick = (user: string) => {
-    const { onSelectUser, clearOnSelect } = this.props;
+  const handleResultClick = (user: string) => {
     onSelectUser(user);
-    this.setState({
-      newUser: clearOnSelect ? "" : user,
-      userSearchResults: [],
-    });
+    setNewUser(clearOnSelect ? "" : user);
+    setUserSearchResults([]);
   };
 
-  handleError = (error: any) => {
-    toast.error(<ToastMsg title="Error" message={error.message} />, {
-      toastId: "error",
-    });
+  const reset = () => {
+    setUserSearchResults([]);
+    setSelectedIndex(-1);
   };
 
-  render() {
-    const { newUser, userSearchResults } = this.state;
-    const { placeholder } = this.props;
-    return (
-      <>
-        <input
-          id="user-name-search"
-          type="text"
-          className="form-control"
-          name="newUser"
-          onChange={this.handleInputChange}
-          placeholder={placeholder}
-          value={newUser}
-          aria-haspopup={Boolean(userSearchResults?.length)}
-        />
-        <div className="search-dropdown">
-          {userSearchResults?.map((user) => {
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (event.key === "ArrowDown") {
+      setSelectedIndex((prevIndex) =>
+        prevIndex < userSearchResults.length - 1 ? prevIndex + 1 : prevIndex
+      );
+    } else if (event.key === "ArrowUp") {
+      setSelectedIndex((prevIndex) =>
+        prevIndex > 0 ? prevIndex - 1 : prevIndex
+      );
+    } else if (event.key === "Enter" && selectedIndex >= 0) {
+      handleResultClick(userSearchResults[selectedIndex].user_name);
+      reset();
+    }
+  };
+
+  React.useEffect(() => {
+    if (!newUser) {
+      return;
+    }
+    throttledSearchUsers();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [newUser]);
+
+  React.useEffect(() => {
+    if (selectedIndex >= 0 && dropdownRef.current) {
+      const option = dropdownRef.current.options[selectedIndex];
+      option.scrollIntoView({ block: "nearest" });
+    }
+  }, [selectedIndex]);
+
+  return (
+    <div
+      className="input-group user-search"
+      style={{
+        width: "100%",
+      }}
```

Comment:
The css class `.input-group-flex` is helpful here:
```suggestion
      className="input-group input-group-flex user-search"
```